bin/packages/build-types.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable eslint-comments/disable-enable-pair, no-console */
+
+/**
+ * External dependencies
+ */
+const execa = require( 'execa' );
+const glob = require( 'fast-glob' );
+const path = require( 'path' );
+
+const tscPath = path.resolve(
+	__dirname,
+	'..',
+	'..',
+	'node_modules',
+	'.bin',
+	'tsc'
+);
+
+async function main() {
+	const packagesWithTs = await glob( 'packages/*/tsconfig.json', {
+		cwd: path.resolve( __dirname, '..', '..' ),
+	} );
+	const projectPaths = packagesWithTs.map( ( tsconfigPath ) =>
+		path.dirname( tsconfigPath )
+	);
+
+	const args = [ '--build', ...projectPaths, ...process.argv.slice( 2 ) ];
+
+	try {
+		await execa( tscPath, args, { stdio: 'inherit' } );
+	} catch {
+		process.exitCode = 1;
+	}
+}
+
+main().catch( ( err ) => {
+	throw err;
+} );
```

Comment:
Is the intention with this to account for Node warning about uncaught errors not exiting with non-zero status code in a future release? I wonder if that's any different if the `throw` takes place in `catch` (still part of the promise). Have you tested it?

Also, if we're handling _some_ errors in `main`, why not handle either all or no potential errors?

Reply:
This isn't doing anything meaningful. I've made this all sync and removed try/catch to let errors surface.